client/webserver/site/src/js/mmutil.ts

```diff
@@ -0,0 +1,923 @@
+    const quoteFeeReserves = quoteFees.estimated.swap + Math.max(quoteFees.estimated.redeem + quoteFees.estimated.refund)
+    let baseBookingFeesPerLot = baseFees.max.swap
+    if (quoteFeeID === baseFeeID) baseBookingFeesPerLot += quoteFees.max.redeem
+    let quoteBookingFeesPerLot = quoteFees.max.swap
+    if (baseFeeID === quoteFeeID) quoteBookingFeesPerLot += baseFees.max.redeem
+    const fees: { base: FeeEstimates, quote: FeeEstimates } = {
+      base: {
+        ...baseFees,
+        bookingFeesPerLot: baseBookingFeesPerLot,
+        bookingFees: baseBookingFeesPerLot * commit.dex.base.lots,
+        feesPerSwap: baseFeeReserves
+      },
+      quote: {
+        ...quoteFees,
+        bookingFeesPerLot: quoteBookingFeesPerLot,
+        bookingFees: quoteBookingFeesPerLot * commit.dex.quote.lots,
+        feesPerSwap: quoteFeeReserves
+      }
+    }
+
+    return { commit, fees }
+  }
+
+  /*
+   * projectedAllocations calculates the required asset allocations from the
+   * user's configuration settings and the current market state.
+   */
+  projectedAllocations () {
+    const {
+      cfg: { uiConfig: { quoteConfig, baseConfig } },
+      baseFactor, quoteFactor, baseID, quoteID, lotSizeConv, quoteLotConv,
+      baseFeeFactor, quoteFeeFactor, baseFeeID, quoteFeeID, baseToken,
+      quoteToken, cexName
+    } = this
+    const { commit, fees } = this.feesAndCommit()
+
+    const bProj = emptyProjection()
+    const qProj = emptyProjection()
+
+    bProj.book = commit.dex.base.lots * lotSizeConv
+    qProj.book = commit.cex.base.lots * quoteLotConv
+
+    bProj.orderReserves = Math.max(commit.cex.base.val, commit.dex.base.val) * baseConfig.orderReservesFactor / baseFactor
+    qProj.orderReserves = Math.max(commit.cex.quote.val, commit.dex.quote.val) * quoteConfig.orderReservesFactor / quoteFactor
+
+    if (cexName) {
+      bProj.cex = commit.cex.base.lots * lotSizeConv
+      qProj.cex = commit.cex.quote.lots * quoteLotConv
+    }
+
+    bProj.bookingFees = commit.dex.base.lots * fees.base.bookingFeesPerLot / baseFeeFactor
+    qProj.bookingFees = commit.dex.quote.lots * fees.quote.bookingFeesPerLot / quoteFeeFactor
+
+    if (baseToken) bProj.swapFeeReserves = fees.base.feesPerSwap * baseConfig.swapFeeN / baseFeeFactor
+    if (quoteToken) qProj.swapFeeReserves = fees.quote.feesPerSwap * quoteConfig.swapFeeN / quoteFeeFactor
+    qProj.slippageBuffer = (qProj.book + qProj.cex + qProj.orderReserves) * quoteConfig.slippageBufferFactor
+
+    const alloc: Record<number, number> = {}
+    alloc[baseID] = Math.round((bProj.book + bProj.cex + bProj.orderReserves) * baseFactor)
+    alloc[baseFeeID] = (alloc[baseFeeID] ?? 0) + Math.round((bProj.bookingFees + bProj.swapFeeReserves) * baseFeeFactor)
+    alloc[quoteID] = Math.round((qProj.book + qProj.cex + qProj.orderReserves + qProj.slippageBuffer) * quoteFactor)
+    alloc[quoteFeeID] = (alloc[quoteFeeID] ?? 0) + Math.round((qProj.bookingFees + qProj.swapFeeReserves) * quoteFactor)
```

Comment:
Some of these asset IDs might be the same. If you're trading `ETH / USDC.ETH`, then `baseID` and `quoteFeeID` would be the same, or if you're trading `USDT.ETH / USDC.ETH` then `baseFeeID` and `quoteFeeID` would be the same.

Reply:
That's what the `(alloc[baseFeeID] ?? 0) +` accounts for. If `baseID === baseFeeID`, it will properly sum them.

Reply:
The first pic is what I selected in `mmsettings` and the second is the result. The fees for only one of the bots is accounted for. 

<img width="949" alt="Screenshot 2024-06-14 at 2 19 10 PM" src="https://github.com/decred/dcrdex/assets/6186350/10f37a2d-7de6-4fae-bf57-d22116420a15">


<img width="463" alt="Screenshot 2024-06-14 at 2 19 32 PM" src="https://github.com/decred/dcrdex/assets/6186350/8eef25ed-a54c-4e0e-9c19-29970415a929">